src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs

```diff
@@ -80,13 +80,21 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
 
             InitializeTestState(test, stateProvider, specProvider);
 
-            BlockHeader header = new(test.PreviousHash, Keccak.OfAnEmptySequenceRlp, test.CurrentCoinbase,
-                test.CurrentDifficulty, test.CurrentNumber, test.CurrentGasLimit, test.CurrentTimestamp, Array.Empty<byte>());
+            BlockHeader header = new(
+                test.PreviousHash,
+                Keccak.OfAnEmptySequenceRlp,
+                test.CurrentCoinbase,
+                test.CurrentDifficulty,
+                test.CurrentNumber,
+                test.CurrentGasLimit,
+                test.CurrentTimestamp,
+                Array.Empty<byte>());
             header.BaseFeePerGas = test.Fork.IsEip1559Enabled ? test.CurrentBaseFee ?? _defaultBaseFeeForStateTest : UInt256.Zero;
             header.StateRoot = test.PostHash;
             header.Hash = header.CalculateHash();
             header.IsPostMerge = test.CurrentRandom is not null;
             header.MixHash = test.CurrentRandom;
+            header.WithdrawalsRoot = test.CurrentWithdrawalsRoot;
```

Comment:
What do we do with:

- `test.CurrentBeaconRoot`
- `test.ParentBlobGasUsed`
- `test.ParentExcessBlobGas`

Reply:
I've tried adding the following lines:

```C#
header.ParentBeaconBlockRoot = test.CurrentBeaconRoot;
// ...
if (test.ParentBlobGasUsed is not null && test.ParentExcessBlobGas is not null)
{
    BlockHeader parent = new(
        parentHash: null!,
        unclesHash: Keccak.OfAnEmptySequenceRlp,
        beneficiary: test.CurrentCoinbase,
        difficulty: test.CurrentDifficulty,
        number: test.CurrentNumber - 1,
        gasLimit: test.CurrentGasLimit,
        timestamp: test.CurrentTimestamp,
        extraData: Array.Empty<byte>()
    )
    {
        BlobGasUsed = (ulong)test.ParentBlobGasUsed,
        ExcessBlobGas = (ulong)test.ParentExcessBlobGas,
    };
    header.ExcessBlobGas = BlobGasCalculator.CalculateExcessBlobGas(parent, spec);
}
```

The intent was to do the same thing as geth is doing:

https://github.com/ethereum/go-ethereum/blob/e9f59b5d5ea44df607f826c72f42916b552ab33d/cmd/evm/internal/t8ntool/execution.go#L167-L180

Unfortunately, this does not produce any observable change to the failing tests: adding or not these lines doesn't change the actual state root (`0xfb267ebc2b5a092a18b6e3828b4ae2c57ece236db0b5afcdab0ddd233ab1d983`), which is not the expected one (`0x52196f6444dac6b01163ab7f09586ce9b5a961c271708b5a3fb3d5a862266722`).

Maybe this extra data is irrelevant and the issue is somewhere else?

Reply:
Fixed: we were not properly deserializing the `Transaction` (missing fields) and we were computing the wrong `TxType`.